python/ray/autoscaler/v2/instance_manager/ray_installer.py

```diff
@@ -49,7 +51,17 @@ def install_ray(self, instance: Instance, head_node_ip: str) -> bool:
                 head_node_ip,
             ),
             setup_commands=with_head_node_ip(setup_commands, head_node_ip),
-            ray_start_commands=with_head_node_ip(ray_start_commands, head_node_ip),
+            # This will prepend envs to the begin of the ray start commands, e.g.
+            # `export RAY_HEAD_IP=<head_node_ip>; \
+            #  export RAY_cloud_instance_id=<instance_id>; \
+            #  ray start --head ...`
+            ray_start_commands=with_envs(
+                ray_start_commands,
+                {
+                    "RAY_HEAD_IP": head_node_ip,
+                    "RAY_cloud_instance_id": instance.instance_id,
```

Comment:
This is a bit confusing because this RAY_env var usually is applied across all components. 

Should we just instead use a special env var like RAY_CLOUD_INSTNCE_ID?

Reply:
> This is a bit confusing because this RAY_env var usually is applied across all components.

wdym by this? I thought there are also configs that are just specific to one component, e.g. GCS related configs.

But I do agree maybe a special ENV could also work.

Reply:
theoretically, if you have RAY_env, on a head node, all the worker nodes will have the same config by default. 

If you set this every node, it is fine, but I feel like it may be prone to error.

Reply:
https://github.com/ray-project/ray/blob/21f3c8096c25e30f8dcfa4a7d99a894786a5c13f/src/ray/raylet/main.cc#L147

Reply:
Ah this is good to know - I think i will go with just having a special env.